frame/evm/src/runner/stack.rs

```diff
@@ -244,6 +192,51 @@ impl<T: Config> Runner<T> {
 impl<T: Config> RunnerT<T> for Runner<T> {
 	type Error = Error<T>;
 
+	fn validate(
+		source: H160,
+		target: Option<H160>,
+		input: Vec<u8>,
+		value: U256,
+		gas_limit: u64,
+		max_fee_per_gas: Option<U256>,
+		max_priority_fee_per_gas: Option<U256>,
+		nonce: Option<U256>,
+		access_list: Vec<(H160, Vec<H256>)>,
+		is_transactional: bool,
+		evm_config: &evm::Config,
+	) -> Result<(), RunnerError<Self::Error>> {
+		let (base_fee, mut weight) = T::FeeCalculator::min_gas_price();
+		let (source_account, inner_weight) = Pallet::<T>::account_basic(&source);
+		weight = weight.saturating_add(inner_weight);
+
+		let _ = fp_evm::CheckEvmTransaction::<Self::Error>::new(
+			fp_evm::CheckEvmTransactionConfig {
+				evm_config,
+				block_gas_limit: T::BlockGasLimit::get(),
+				base_fee,
+				chain_id: T::ChainId::get(),
+				is_transactional,
+			},
+			fp_evm::CheckEvmTransactionInput {
+				chain_id: Some(T::ChainId::get()),
+				to: target,
+				input,
+				nonce: nonce.unwrap_or(source_account.nonce),
+				gas_limit: gas_limit.into(),
+				gas_price: None,
+				max_fee_per_gas,
+				max_priority_fee_per_gas,
+				value,
+				access_list,
+			},
+		)
+		.validate_in_block_for(&source_account)
```

Comment:
Did you miss `with_chain_id` here? You check it in the ethereum pallet.

Reply:
In `pallet-ethereum` we need it because the `chain_id` is part of the user input / signature. The runner just uses the `ChainId` owned by `pallet-evm` so there is no need to validate it.

Reply:
@AsceticBear wait, even if we pass the `chain_id` to the runner when coming from `pallet-ethereum`, the validation is skipped in the runner.

Reply:
Could you please explain why it is skipped in the runner?

Reply:
> Could you please explain why it is skipped in the runner?

`call`, `create` and `create2` methods signature changed to accept a `validate` boolean parameter. This means that calls to the Runner that are already validated - like in the case of `pallet-ethereum` - don't need to do the checks twice.

Reply:
Okay